pynws/nws.py

```diff
@@ -17,56 +22,75 @@
 class NwsError(Exception):
     """Error in Nws Class"""
 
-    def __init__(self, message):
+    def __init__(self: NwsError, message: str):
         super().__init__(message)
         self.message = message
 
 
 class Nws:
     """Class to more easily get data for one location."""
 
-    def __init__(self, session, userid, latlon=None, station=None):
-        self.session = session
-        self.latlon = latlon
-        self.station = station
-        self.userid = userid
-
-        self.wfo = None
-        self.x = None
-        self.y = None
-
-        self.forecast_zone = None
-        self.county_zone = None
-        self.fire_weather_zone = None
-
-    async def get_points_stations(self):
+    def __init__(
+        self: Nws,
+        session: ClientSession,
+        userid: str,
+        latlon: Optional[Tuple[float, float]] = None,
+        station: Optional[str] = None,
+    ):
+        if not session:
+            raise NwsError(f"{session!r} is required")
+        if not isinstance(userid, str) or not userid:
+            raise NwsError(f"{userid!r} is required")
+        if latlon and (not isinstance(latlon, tuple) or len(latlon) != 2):
```

Comment:
Thinking about this more.  The default for `latlon` to be `None` was originally motivated so that just a station could be supplied.  But the same could be said for the `forecast_zone`, etc, which was added later.  However support for initializing this class with those values wasn't implemented.  It probably makes more sense to completely remove the station here, but allow it to be set via a setter/getter or method?  Same for the `*_zones`.  Lets' keep this out of this PR though.